src/dotnet-install.ps1

```diff
@@ -204,6 +204,15 @@ function Get-Machine-Architecture() {
         return $ENV:PROCESSOR_ARCHITEW6432
     }
 
+    # covers the case when PS x64 is run on ARM machine
+    if( (wmic os get osarchitecture)[2] -like "ARM*" ) {
```

Comment:
Apparently, [wmic is deprecated](https://docs.microsoft.com/en-us/windows/win32/wmisdk/wmic) starting Windows 10 21H1.
Arm windows machines are available starting Windows 11.
It may be better to use the [recommended approach](https://docs.microsoft.com/en-us/powershell/scripting/learn/ps101/07-working-with-wmi?view=powershell-7.2) to avoid technical dept.